src/kits/tracker/ContainerWindow.cpp

```diff
@@ -456,48 +444,16 @@ DraggableContainerIcon::MouseMoved(BPoint where, uint32, const BMessage*)
 
 
 void
-DraggableContainerIcon::FrameMoved(BPoint)
-{
-	BMenuBar* bar = dynamic_cast<BMenuBar*>(Parent());
-	ThrowOnAssert(bar != NULL);
-
-	// TODO: ugly hack following:
-	// This is a trick to get the actual width of all menu items
-	// (BMenuBar may not have set the item coordinates yet...)
-	float width, height;
-	uint32 resizingMode = bar->ResizingMode();
-	bar->SetResizingMode(B_FOLLOW_NONE);
-	bar->GetPreferredSize(&width, &height);
-	bar->SetResizingMode(resizingMode);
-
-	//BMenuItem* item = bar->ItemAt(bar->CountItems() - 1);
-	//if (item == NULL)
-	//	return;
-
-	// BeOS shifts the coordinates for hidden views, so we cannot
-	// use them to decide if we should be visible or not...
-
-	float gap = bar->Frame().Width() - 2 - width; //item->Frame().right;
-
-	if (gap <= Bounds().Width() && !IsHidden())
-		Hide();
-	else if (gap > Bounds().Width() && IsHidden())
-		Show();
-}
-
-
-void
 DraggableContainerIcon::Draw(BRect updateRect)
 {
 	BContainerWindow* window = dynamic_cast<BContainerWindow*>(Window());
 	ThrowOnAssert(window != NULL);
 
-	if (be_control_look != NULL) {
-		BRect rect(Bounds());
-		rgb_color base = ui_color(B_MENU_BACKGROUND_COLOR);
-		be_control_look->DrawMenuBarBackground(this, rect, updateRect, base,
-			0, 0);
-	}
+	BRect rect(Bounds());
+	be_control_look->DrawBorder(this, rect, updateRect, fColor,
+		B_PLAIN_BORDER, 0, BControlLook::B_BOTTOM_BORDER);
+	be_control_look->DrawMenuBarBackground(this,
+		rect, updateRect, fColor);
```

Comment:
I think I misunderstood how this code worked, and changed here as a result of misunderstanding and forgot to put it back. Will do.